core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/ast/planNodes/DatatypeFilter.java

```diff
@@ -17,10 +17,12 @@
 public class DatatypeFilter extends FilterPlanNode {
 
 	private final Resource datatype;
+	private StackTraceElement[] stackTrace;
 
 	public DatatypeFilter(PlanNode parent, Resource datatype) {
 		super(parent);
 		this.datatype = datatype;
+//		stackTrace = Thread.currentThread().getStackTrace();
 	}
 
```

Comment:
The stack trace is used while debugging, but I'm not sure how to make it so that it is only populated when we are debugging (or asserts enabled). So I've just commented them out instead.

Reply:
This is fine for me, but for future reference you can do `if (logger.isDebugEnabled())`  to check that we're logging at debug level (which is usually a good proxy for actually doing debugging :)).